### PR TITLE
solve Invalid Email Format in Text Field #245

### PR DESCRIPTION
**Pull request by DevJethava**
solve Bug Report: Accepting Invalid Email Format in Text Field #245

### CODE REVIEW

**Comment by hieuwu** [comment]
Hey can you please provide some screenshots? It would be very helpful when I review the PR

**Comment by DevJethava** [comment]
Email cheker
<p>
Signup screen
signing screen
</p>

File: app/src/main/java/com/hieuwu/groceriesstore/presentation/authentication/signin/SignInViewModel.kt [hunk 1]
```diff
@@ -40,6 +40,10 @@ class SignInViewModel @Inject constructor(
         _password.value = newPassword
     }
 
+    fun isValidEmail(): Boolean {
+        return android.util.Patterns.EMAIL_ADDRESS.matcher(_email.value).matches()
```
Comment (hieuwu) [inline app/src/main/java/com/hieuwu/groceriesstore/presentation/authentication/signin/SignInViewModel.kt:44]: Please import thí one for simplicity

```suggestion
import android.util.Patterns.EMAIL_ADDRESS
        return EMAIL_ADDRESS.matcher(_email.value).matches()
```

File: app/src/main/java/com/hieuwu/groceriesstore/presentation/authentication/signin/SignInScreen.kt [hunk 2]
```diff
@@ -127,7 +130,15 @@ fun SignInScreen(
                 )
                 Button(
                     modifier = Modifier.fillMaxWidth(),
-                    onClick = { viewModel.signIn() },
+                    onClick = {
+                        if (viewModel.isValidEmail())
+                            viewModel.signIn()
+                        else {
+                            CoroutineScope(Dispatchers.Main).launch {
```
Comment (hieuwu) [inline app/src/main/java/com/hieuwu/groceriesstore/presentation/authentication/signin/SignInScreen.kt:137]: Please use remember coroutineScope to handle this

Something like:
```kotlin
// In composable scope
val coroutine = rememberCoroutineScope()

...
coroutineScope.launch {
...
}
```

**Comment by DevJethava** [comment]
@hieuwu, could you please check PR and Marge? I appreciate and consider your comments as I start my career as an open-source contributor. Thanks.

**Review by hieuwu — APPROVED** [review]

**Pull request merged**